ledger-core/virtual/object/object_migration_test.go

```diff
@@ -109,21 +115,29 @@ func TestSMObject_MigrationCreateStateReport_IfStateIsEmptyAndNoCounters(t *test
 		UnpublishAllMock.Return().
 		ShareMock.Set(
 		func(data interface{}, flags smachine.ShareDataFlags) (s1 smachine.SharedDataLink) {
-			require.IsType(t, &payload.VStateReport{}, data)
+			switch d := data.(type) {
+			case payload.VStateReport:
+				require.IsType(t, &payload.VStateReport{}, d)
+			}
 			require.Equal(t, smachine.ShareDataFlags(0), flags)
 			sharedData = smachine.NewUnboundSharedData(data)
 			return sharedData
 		}).
 		PublishMock.Set(
 		func(key interface{}, data interface{}) (b1 bool) {
-			refKey := finalizedstate.BuildReportKey(smObject.Reference)
-			require.Equal(t, refKey, key)
-			require.Equal(t, sharedData, data)
+			switch d := data.(type) {
+			case finalizedstate.ReportKey:
+				refKey := finalizedstate.BuildReportKey(smObject.Reference)
```

Comment:
make that checks consistent with each other

Reply:
Fixed